Geometry/HGCalCommonData/plugins/HGCalParametersESModule.cc

```diff
@@ -13,40 +16,66 @@
 class HGCalParametersESModule : public edm::ESProducer {
 public:
   HGCalParametersESModule(const edm::ParameterSet&);
-  ~HGCalParametersESModule(void) override;
 
   using ReturnType = std::unique_ptr<HGCalParameters>;
 
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
   ReturnType produce(const IdealGeometryRecord&);
 
 private:
-  std::string name_, namew_, namec_, namet_;
-  edm::ESGetToken<DDCompactView, IdealGeometryRecord> cpvToken_;
+  std::string name_, name2_, namew_, namec_, namet_;
+  bool fromDD4Hep_;
+  edm::ESGetToken<DDCompactView, IdealGeometryRecord> cpvTokenDDD_;
+  edm::ESGetToken<cms::DDCompactView, IdealGeometryRecord> cpvTokenDD4Hep_;
 };
 
 HGCalParametersESModule::HGCalParametersESModule(const edm::ParameterSet& iC) {
-  name_ = iC.getUntrackedParameter<std::string>("Name");
-  namew_ = iC.getUntrackedParameter<std::string>("NameW");
-  namec_ = iC.getUntrackedParameter<std::string>("NameC");
-  namet_ = iC.getUntrackedParameter<std::string>("NameT");
+  name_ = iC.getUntrackedParameter<std::string>("name");
+  name2_ = iC.getUntrackedParameter<std::string>("name2");
+  namew_ = iC.getUntrackedParameter<std::string>("nameW");
+  namec_ = iC.getUntrackedParameter<std::string>("nameC");
+  namet_ = iC.getUntrackedParameter<std::string>("nameT");
+  fromDD4Hep_ = iC.getParameter<bool>("fromDD4Hep");
 #ifdef EDM_ML_DEBUG
   edm::LogVerbatim("HGCalGeom") << "HGCalParametersESModule for " << name_ << ":" << namew_ << ":" << namec_ << ":"
-                                << namet_;
+                                << namet_ << " and fromDD4Hep flag " << fromDD4Hep_;
 #endif
   auto cc = setWhatProduced(this, name_);
-  cpvToken_ = cc.consumes<DDCompactView>(edm::ESInputTag{});
+  cpvTokenDDD_ = cc.consumes<DDCompactView>(edm::ESInputTag());
+  cpvTokenDD4Hep_ = cc.consumesFrom<cms::DDCompactView, IdealGeometryRecord>(edm::ESInputTag());
 }
 
-HGCalParametersESModule::~HGCalParametersESModule() {}
+void HGCalParametersESModule::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.addUntracked<std::string>("name", "HGCalEESensitive");
```

Comment:
Is there a reason why these are untracked? It seems like they would be better as tracked parameters.

Reply:
Done